Integration/AndroidAAR/BabylonNative/src/main/java/BabylonNative/BabylonView.java

```diff
@@ -0,0 +1,113 @@
+package BabylonNative;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.MotionEvent;
+import android.view.SurfaceHolder;
+import android.view.SurfaceView;
+import android.view.View;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+
+public class BabylonView extends SurfaceView implements SurfaceHolder.Callback2, View.OnTouchListener {
+    private static final String TAG = "BabylonView";
+    private boolean mViewReady = false;
+    private ViewDelegate mViewDelegate;
+
+    public BabylonView(Context context, ViewDelegate viewDelegate) {
+        super(context);
+        init(viewDelegate);
+        BabylonNative.Wrapper.initEngine(context.getResources().getAssets());
+    }
+
+    private void init(ViewDelegate viewDelegate) {
```

Comment:
I would personally just move this into the constructor to make it clear that all this happens only at construction time.

Reply:
sounds good.